openedx/openedx-platform
# Fix: Add ContentErrorData for Course Outline DAGs

## Description

This PR fixes Course Outline DAGs with duplicate sequences. Previously when a course outline had duplicate sequences, the outline would not generate and raise a ValueError. There were no checks for duplicate sequences before the generation of the course outline because it is not possible to create duplicate sequences in Studio, but is possible when a Course Author imports a course. Now before the course outline is generated, it will be checked for duplicate sequences. If a duplicate sequence is found an error will be logged for Partner Support to see in the Django Admin and the duplicate will be deleted. This change will impact the Course Author.

## Supporting information

Jira issue: [TNL-8208](https://openedx.atlassian.net/browse/TNL-8208)

## Testing instructions

- In Studio, open an existing course or create a new course.
- Go to Tools>Import and import [this OLX](https://github.com/edx/edx-platform/files/6728409/ex_dag1.tar.gz).
- Check that a course outline was generated with [Django Admin](http://localhost:18000/admin/learning_sequences/coursecontext).
    -  _**Need to be connected to VPN**_
- Check that there is a Content Error logged with the course outline.

## Deadline

None

## Other information

None
